03-event-emitters/3-task/json-writer.ts

```diff
@@ -0,0 +1,35 @@
+import * as fs from "node:fs";
+import path from "node:path";
+
+class JSONWriter {
+  outputPath: string;
+
+  constructor(outputPath: string) {
+    this.outputPath = outputPath;
+  }
+
+  createDir(dirPath: string) {
+    if (!fs.existsSync(dirPath)) {
+      fs.mkdirSync(dirPath, { recursive: true });
+    }
+  }
+
+  write(data: string[]) {
+    return new Promise((resolve, reject) => {
+      const dirPath = path.dirname(this.outputPath);
+      this.createDir(dirPath);
+
+      const writeStream = fs.createWriteStream(this.outputPath);
+      writeStream.on("error", (error: Error) =>
+        reject(`Error writing to TXT file: ${error}`)
+      );
+
+      data.forEach((line) => writeStream.write(line + "\n"));
```

Comment:
One small thing is the use of maginc strings: "\n".

The more important here is again, the usage of pipes, which can simplify your code:
![image](https://github.com/molek95/nodejs-global-mentoring/assets/39033848/5eabcc7f-7870-45df-8039-67d6b4f8dd03)